### DISCUSSION TITLE
Custom GPU Targets

### DISCUSSION THREAD

**Discussion by tpkessler** [rocm-arch/rocm-arch#695] [Ideas]
This is a follow-up of PR #650.

> Which packages were you having trouble building without the default targets? I didn't have any issues.

I tried it with `rccl` and `hip-runtime-amd`. Both packages did not build in a clean chroot.

> It looks more redundant but it lowers the maintenance requirements.

That's a very good idea. I would write it in the following way:
```bash
build() {
  cmake_flags=('-DCMAKE_INSTALL_PREFIX=/opt/rocm'
        '-Drocprim_DIR=/opt/rocm/rocprim/rocprim/lib/cmake/rocprim'
        '-DBUILD_CLIENTS_SAMPLES=OFF')
  if [[ -n "$AMDGPU_TARGETS" ]]; then
      cmake_flags+=("-DAMDGPU_TARGETS=$AMDGPU_TARGETS")
  fi

  CXX=/opt/rocm/bin/hipcc \
  CXXFLAGS="${CXXFLAGS} -fcf-protection=none" \
  cmake -Wno-dev -S "$_dirname" \
        "${cmake_flags[@]}"
  make
}
```

**Comment by ghost** [comment]
> I tried it with `rccl` and `hip-runtime-amd`. Both packages did not build in a clean chroot.

Good point. I didn't try building them in a clean chroot.

Anyways, I think your idea with the Bash array is the best since hardcoding the targets is a massive pain for maintenance.

**Reply by acxz** [reply]
As a side, @Ta180m what AUR helper do you use?

We sort of expect the PRs to be tested in chroots before being submitted. See: https://github.com/rocm-arch/rocm-arch#contributing (maybe I should add a PR template)

Recently, [`paru`](https://aur.archlinux.org/packages/paru) added the ability to build AUR packages/local PKGBUILDs in a clean chroot which makes life quite easy than the previous more manual methods.

Take a look at the `--chroot` flag for paru.

aurutils is another helper that has devtool features as well.

**Reply by ghost** [reply]
I use `paru`, so I will try using the `--chroot` flag.

**Reply by tpkessler** [reply]
I've added the new logic to `rocsparse 5.1.0`.
